### PR TITLE
 fruit shop

### PR DESCRIPTION
**Pull request by Dinexpod**

### CODE REVIEW

File: src/main/java/shop/Main.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        System.out.println("Avaliable fruits: " + main.getAvailableFruits("2018.10.12"));
+
+        System.out.println("Spoiled fruits (SOME TYPE): " + main.getSpoiledFruit("2018.10.12", TypeFruit.BANANA));
+        System.out.println("Avaliable fruits (SOME TYPE): " + main.getAvailableFruits("2018.10.12", TypeFruit.AVOCADO));
+
+        System.out.println("EXIST: " + main.getAddedFruits("2018.10.10"));
+        System.out.println("EXIST (SOME TYPE): " + main.getAddedFruits("2018.10.10", TypeFruit.GRAPE));
+    }
+
+    private List<Fruit> getAddedFruits(String date) {
```
Comment (boroda4436) [inline src/main/java/shop/Main.java]: @Dinexpod Why you need these methods?? why we can't just call tradeShop.getAddedFruits(date) ?

File: src/main/java/shop/Main.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+    private List<Fruit> getSpoiledFruit(String date) {
+        return tradeShop.getSpoiledFruits(parseDate(date));
+    }
+
+    private List<Fruit> getSpoiledFruit(String date, TypeFruit type) {
+        return tradeShop.getSpoiledFruits(parseDate(date), type);
+    }
+
+    private Date parseDate(String date) {
```
Comment (boroda4436) [inline src/main/java/shop/Main.java]: String date?

File: src/main/java/shop/Main.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    private void doSave() {
+        try {
+            tradeShop.save("src/main/java/shop/allInformation.json");
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void addToList(Fruit fruit) {
```
Comment (boroda4436) [inline src/main/java/shop/Main.java]: Redundant level of encapsulation in all methods where you are calling tradeShop methods

File: src/main/java/shop/Main.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            tradeShop.addFruit(createPath());
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static String createPath() {
+        Date date = new Date();
+        SimpleDateFormat formatForDateNow = new SimpleDateFormat("yyyy.MM.dd");
+        return ("src/main/java/shop/" + "(" + countOfPosts++ + ")" + formatForDateNow.format(date) + ".json");
```
Comment (boroda4436) [inline src/main/java/shop/Main.java]: please, do countOfPosts++ in a separate line above

File: src/main/java/shop/TradeShop.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        doPrintWriter(allPostages, file);
+    }
+
+    void load(String pathToJsonFile) {
+        fruits = null;
+        File file = new File(pathToJsonFile);
+        ObjectMapper mapper = new ObjectMapper();
+
+        try {
+            fruits = Arrays.asList(mapper.readValue(file, Fruit[].class));
```
Comment (boroda4436) [inline src/main/java/shop/TradeShop.java]: lets see. You already have removed your data from fruits. What if you will receive exception here? How you program will work without data???

File: src/main/java/shop/Main.java (hunk 1 above)
Comment (Dinexpod) [reply src/main/java/shop/Main.java]: Знаєш, оце відпочив годинку а потім сів знову за код і дійсно подумав навіщо я поробив оці методи, памятаю бо щось не виходило а потім переробив і все норм, працює... хм... глюк

File: src/main/java/shop/Main.java (hunk 3 above)
Comment (Dinexpod) [reply src/main/java/shop/Main.java]: ну да)

File: src/main/java/shop/Main.java (hunk 4 above)
Comment (Dinexpod) [reply src/main/java/shop/Main.java]: ok

File: src/main/java/shop/TradeShop.java (hunk 5 above)
Comment (Dinexpod) [reply src/main/java/shop/TradeShop.java]: ok, changed

File: src/main/java/shop/Main.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        Fruit apple = new Fruit(TypeFruit.APPLE, 1, (new Date()), 117);
+        Fruit avocado = new Fruit(TypeFruit.AVOCADO, 3, (new Date()), 19);
+        Fruit grape = new Fruit(TypeFruit.GRAPE, 10, (new Date()), 57);
+
+        tradeShop.add(banana);
+        tradeShop.add(apple);
+        tradeShop.add(avocado);
+        tradeShop.add(grape);
+
+        tradeShop.addFruit(createPath());
```
Comment (boroda4436) [inline src/main/java/shop/Main.java]: Why you need this method _addFruit_ ?

File: src/main/java/shop/Main.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+    public static void main(String[] args) throws IOException {
+        TradeShop tradeShop = new TradeShop();
+
+        Fruit banana = new Fruit(TypeFruit.BANANA, 1, (new Date()), 187);
+        Fruit apple = new Fruit(TypeFruit.APPLE, 1, (new Date()), 117);
+        Fruit avocado = new Fruit(TypeFruit.AVOCADO, 3, (new Date()), 19);
+        Fruit grape = new Fruit(TypeFruit.GRAPE, 10, (new Date()), 57);
+
+        tradeShop.add(banana);
```
Comment (boroda4436) [inline src/main/java/shop/Main.java]: No, where is method void addFruits(List<Fruit> fruits, String pathToJsonFile) ???
I have updated requirements in channel

File: src/main/java/shop/Main.java (hunk 6 above)
Comment (Dinexpod) [reply src/main/java/shop/Main.java]: changed name of method

File: src/main/java/shop/Main.java (hunk 7 above)
Comment (Dinexpod) [reply src/main/java/shop/Main.java]: changed

**Pull request merged**